pkg/externalendpoint/externalendpoint.go

```diff
@@ -0,0 +1,81 @@
+package externalendpoint
+
+import (
+	"os"
+	"path/filepath"
+
+	externalHTTPEndpointV1alpha1 "github.com/dapr/dapr/pkg/apis/externalHTTPEndpoint/v1alpha1"
+	"github.com/dapr/dapr/utils"
+	"github.com/dapr/kit/logger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	externalEndpointKind = "ExternalHTTPEndpoint"
+)
+
+// LoadLocalExternalEndpoint loads external endpoint configurations from local folders.
+func LoadLocalExternalEndpoint(log logger.Logger, runtimeID string, paths ...string) []*externalHTTPEndpointV1alpha1.ExternalHTTPEndpoint {
+	configs := []*externalHTTPEndpointV1alpha1.ExternalHTTPEndpoint{}
+	for _, path := range paths {
+		loaded := loadLocalExternalHTTPEndpointPath(log, runtimeID, path)
+		if len(loaded) > 0 {
+			configs = append(configs, loaded...)
+		}
+	}
+	return configs
+}
+
+func loadLocalExternalHTTPEndpointPath(log logger.Logger, runtimeID string, path string) []*externalHTTPEndpointV1alpha1.ExternalHTTPEndpoint {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return nil
+	}
+
+	files, err := os.ReadDir(path)
+	if err != nil {
+		log.Errorf("Failed to read external http endpoint files from path %s: %v", path, err)
+		return nil
+	}
+
+	configs := make([]*externalHTTPEndpointV1alpha1.ExternalHTTPEndpoint, 0, len(files))
+
+	type typeInfo struct {
+		metav1.TypeMeta `json:",inline"`
+	}
+
+	for _, file := range files {
+		if !utils.IsYaml(file.Name()) {
+			log.Warnf("A non-YAML external http endpoint file %s was detected, it will not be loaded", file.Name())
```

Comment:
> but it would be nice to have a single "resource loader" that scans the resources paths and loads, in a single loop, all resources: resiliency policies, components, endpoints…

I agree, and as you said it should be it's own dedicated task outside the scope of this PR.